shared/domains/chelonia/GIMessage.js

```diff
@@ -38,59 +53,97 @@ export class GIMessage {
   static OP_PROTOCOL_UPGRADE: 'pu' = 'pu'
   static OP_PROP_SET: 'ps' = 'ps' // set a public key/value pair
   static OP_PROP_DEL: 'pd' = 'pd' // delete a public key/value pair
+  static OP_CONTRACT_AUTH: 'ca' = 'ca' // authorize a contract
+  static OP_CONTRACT_DEAUTH: 'cd' = 'cd' // deauthorize a contract
+  static OP_ATOMIC: 'at' = 'at' // atomic op
+  static OP_KEYSHARE: 'ks' = 'ks' // key share
+  static OP_KEY_REQUEST: 'kr' = 'kr' // key request
+  static OP_KEY_REQUEST_RESPONSE: 'krr' = 'krr' // key request response
 
   // eslint-disable-next-line camelcase
   static createV1_0 (
-    contractID: string | null = null,
-    previousHEAD: string | null = null,
-    op: GIOp,
-    manifest: string,
-    signatureFn?: Function = defaultSignatureFn
+    {
+      contractID,
+      originatingContractID,
```

Comment:
What is this `originatingContractID`, and which Wiki document discusses it?

Reply:
`contractID` is the contract being modified. `originatingContractID` is the contract initiating the interaction (used for cross-contract communication). In other words, `contractID` is self-referential in a given contract and `originatingContractID` refers to the contract that wrote to it. This is documented in the Key sharing document.